pkg/skaffold/debug/transform.go

```diff
@@ -311,3 +311,37 @@ func describe(obj runtime.Object) (group, version, kind, description string) {
 	}
 	return
 }
+
+// exposePort adds a `ContainerPort` instance or amends an existing entry with the same port.
+func exposePort(entries []v1.ContainerPort, portName string, port int32) []v1.ContainerPort {
+	for i := range entries {
+		// ports and names must be unique so rewrite an existing entry if found
+		if entries[i].ContainerPort == port || entries[i].Name == portName {
```

Comment:
so this might be still ambiguous, depending on the order of the entries..., is that what we want? 

i.e. 
```
portName = "a", port= 80

entries1 = [ 
{a, 100}, ---> we are stopping here
{b,80}
]

entries1 = [ 
{b,80}, --->  we are stopping here
{a, 100}

]
```

Reply:
Good catch!

Reply:
Note that this shouldn't happen since we try to find unallocated ports.  But we may have other language runtimes where that isn't the case.